Realm/ObjectServerTests/RLMObjectServerTests.m

```diff
@@ -88,97 +104,200 @@ - (void)testOpenRealmWithAdminToken {
         // Add objects.
-        [r beginWriteTransaction];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"child-1"]]];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"child-2"]]];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"child-3"]]];
-        [r commitWriteTransaction];
-        [user waitForUploadToFinish:url];
+        [self addSyncObjectsToRealm:r descriptions:@[@"child-1", @"child-2", @"child-3"]];
+        WAIT_FOR_UPLOAD(user, url);
         CHECK_COUNT(3, SyncObject, r);
     }
 }
 
 /// If client B deletes objects from a synced Realm, client A should see the effects of that deletion.
 - (void)testRemoteDeleteObjects {
-    NSURL *url = [NSURL URLWithString:@"realm://localhost:9080/~/testBasicSync"];
+    NSURL *url = REALM_URL();
     RLMSyncUser *user = [self logInUserForCredential:[RLMObjectServerTests basicCredential:self.isParent]
                                               server:[RLMObjectServerTests authServerURL]];
-    NSError *error = nil;
-    RLMRealm *r = [self openRealmForURL:url user:user error:&error];
-    XCTAssertNil(error, @"Error when opening Realm: %@", error);
+    RLMRealm *r = [self openRealmForURL:url user:user];
     if (self.isParent) {
-        [user waitForDownloadToFinish:url];
+        WAIT_FOR_DOWNLOAD(user, url);
         // Add objects.
-        [r beginWriteTransaction];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"parent-1"]]];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"parent-2"]]];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"parent-3"]]];
-        [r commitWriteTransaction];
-        [user waitForUploadToFinish:url];
+        [self addSyncObjectsToRealm:r descriptions:@[@"parent-1", @"parent-2", @"parent-3"]];
+        WAIT_FOR_UPLOAD(user, url);
         CHECK_COUNT(3, SyncObject, r);
         RLMRunChildAndWait();
-        [user waitForDownloadToFinish:url];
+        WAIT_FOR_DOWNLOAD(user, url);
         CHECK_COUNT(0, SyncObject, r);
     } else {
-        [user waitForDownloadToFinish:url];
+        WAIT_FOR_DOWNLOAD(user, url);
         CHECK_COUNT(3, SyncObject, r);
         [r beginWriteTransaction];
         [r deleteAllObjects];
         [r commitWriteTransaction];
-        [user waitForUploadToFinish:url];
+        WAIT_FOR_UPLOAD(user, url);
         CHECK_COUNT(0, SyncObject, r);
     }
 }
 
+#pragma mark - Multiple Realm Sync
+
+/// If a client opens multiple Realms, there should be one session object for each Realm that was opened.
+- (void)testMultipleRealmsSessions {
+    NSURL *urlA = CUSTOM_REALM_URL(@"a");
+    NSURL *urlB = CUSTOM_REALM_URL(@"b");
+    NSURL *urlC = CUSTOM_REALM_URL(@"c");
+    RLMSyncUser *user = [self logInUserForCredential:[RLMObjectServerTests basicCredential:self.isParent]
+                                              server:[RLMObjectServerTests authServerURL]];
+    // Open three Realms.
+    __unused RLMRealm *realmA = [self openRealmForURL:urlA user:user];
```

Comment:
If this is expecting the realms to stay alive until the end of the scope the variables need to be marked with `__attribute__((objc_precise_lifetime))`.